tests/testStatsdClient.cpp

```diff
@@ -31,10 +31,19 @@ void mock(StatsdServer& server, std::vector<std::string>& messages) {
     } while (server.errorMessage().empty() && !messages.back().empty());
 }
 
-void throwOnError(const StatsdClient& client, bool expectEmpty = true, const std::string& extraMessage = "") {
-    if (client.errorMessage().empty() != expectEmpty) {
-        std::cerr << (expectEmpty ? client.errorMessage() : extraMessage) << std::endl;
-        throw std::runtime_error(expectEmpty ? client.errorMessage() : extraMessage);
+template <typename SocketWrapper>
```

Comment:
i templated this so i could use it with the client or the server as they both adhere to the same spec in terms of getting error strings